rasa/shared/core/domain.py

```diff
@@ -1767,6 +1767,23 @@ def required_slots_for_form(self, form_name: Text) -> List[Text]:
 
         return []
 
+    def count_slot_mapping_statistics(self) -> Tuple[int, int]:
+        """Counts the total number of slot mappings and custom slot mappings.
+
+        Returns:
+            A pair of integers where the first entry is the total number of mappings
+            and the second entry is the total number of custom mappings.
+        """
+        total_mappings = 0
+        custom_mappings = 0
+        for slot in self.slots:
+            for mapping in slot.mappings:
+                total_mappings += 1
+                if mapping.get("type") == SlotMapping.CUSTOM:
+                    custom_mappings += 1
+
```

Comment:
I think we should also have a 3rd metric to count the number of slot mappings that include `conditions` key.